src/identity/storage/ResourceStoreStorageAdapterFactory.ts

```diff
@@ -0,0 +1,149 @@
+import type { Adapter, AdapterPayload } from 'oidc-provider';
+import type { ResourceIdentifier } from '../../ldp/representation/ResourceIdentifier';
+import { trimTrailingSlashes } from '../../util/PathUtil';
+import type { KeyValueStore } from './KeyValueStore';
+import { StorageAdapterFactory } from './StorageAdapterFactory';
+
+export interface ResourceStoreStorageAdapterArgs {
+  baseUrl: string;
+  storagePathname: string;
+  store: KeyValueStore;
+}
+
+/**
+ * An IdP storage adapter that uses a ResourceStore
+ * to persist data.
+ */
+export class ResourceStoreStorageAdapter implements Adapter {
```

Comment:
```suggestion
export class IdpResourceStore implements Adapter {
```

Reply:
Again, I disagree. This is not a ResourceStore. It's a storage adapter. Calling it a ResourceStore is false advertising.